server/src/models/segment_model.ts

```diff
@@ -135,8 +127,38 @@ export async function insertSegments(segments: Segment[]) {
       }
     }
 
-    const intersectedSegments = await db.findMany({
-      analyticUnitId,
+    let cache = await AnalyticUnitCache.findById(analyticUnitId);
```

Comment:
```suggestion
    const cache = await AnalyticUnitCache.findById(analyticUnitId);
```

I don't like to make `const` in functions scope. It's silly